src/main.ts

```diff
@@ -284,36 +301,70 @@ function getFetchInfo(ref: string, commit: string): IFetchInfo {
   return result
 }
 
-async function configGitAuth(token: string) {
+async function configGitAuthForSubmodules(token: string, repoDir: string) {
+  // Set authentication
+  const basicCredential = Buffer.from(
+    `x-access-token:${token}`,
+    'utf8'
+  ).toString('base64')
+  core.setSecret(basicCredential)
+
+  await exec.exec(
+    `git submodule foreach --recursive sh -c "git config --local --add 'http.https://github.com/.extraheader' 'AUTHORIZATION: basic ${basicCredential}'"`,
+    [],
+    { cwd: repoDir }
+  )
+  await exec.exec(
+    `git submodule foreach --recursive sh -c "git config --local --add 'url.https://github.com/.insteadOf' 'git@github.com:'"`,
+    [],
+    { cwd: repoDir }
+  )
+}
+
+async function configGitAuth(token: string, global: boolean, repoDir: string) {
   // Set authentication
   const basicCredential = Buffer.from(
     `x-access-token:${token}`,
     'utf8'
   ).toString('base64')
   core.setSecret(basicCredential)
 
+  var configSelector = '--local'
+  if (global) {
+    configSelector = '--global'
+  }
+
   // (NSL-2981) Remove previous extra auth header if any
   await exec.exec(
-    'git config --global --unset-all http.https://github.com/.extraheader',
+    `git config ${configSelector} --unset-all http.https://github.com/.extraheader`,
     [],
-    { ignoreReturnCode: true }
+    { ignoreReturnCode: true, cwd: repoDir ? repoDir : undefined }
```

Comment:
BTW why do we do `cwd: repoDir ? repoDir : undefined` here but just `cwd: repoDir` above in confiGitAuthForSubmodules?

Reply:
In the second case I know for sure that the caller will set `repoDir` to a non empty string. But it's better to be consistent and fix `configGitAuthForSubmodules` as well.